test/functional/pdo_sqlsrv/MsCommon_mid-refactor.inc

```diff
@@ -0,0 +1,1833 @@
+<?php
+/*
+    Microsoft SQL Server Driver for PHP - Unit Test Framework
+    Copyright (c) Microsoft Corporation.  All rights reserved.
+
+    Description:
+        Common functions (shared by all tests).
+
+*/
+
+//
+// looks like an additional file (in addition to pdo_test_base.inc) may be needed for these PHPTs
+// to be runnable from the MSSQL teams' internal proprietary test running system
+//
+require  'MsData.inc';
+
+function isAEQualified($conn)
+{
+    $msodbcsql_ver = $conn->getAttribute(PDO::ATTR_CLIENT_VERSION)["DriverVer"];
+    $server_ver = $conn->getAttribute(PDO::ATTR_SERVER_VERSION);
+    $msodbcsql_maj = explode(".", $msodbcsql_ver)[0];
+    if ($msodbcsql_maj < 17 || explode('.', $server_ver)[0] < 13) {
+        return false;
+    }
+    return true;
+}
+/*
+// TO BE DELETED
+function connect($options=array())
+{
+  try
+  {
+    // simply use $databaseName from MsSetup.inc to facilitate testing in Azure,
+    // which does not support switching databases
+    require  'MsSetup.inc';
+    $conn = new PDO( "sqlsrv:Server=$server;database=$databaseName;ConnectionPooling=false;" , $uid, $pwd, $options);
+    $conn->setAttribute( PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION );
+    create_and_insert_table1($conn);
+    create_and_insert_table2($conn);
+      return $conn;
+  }
+
+  catch( PDOException $e )
+  {
+    var_dump( $e );
+    exit;
+  }
+
+  catch(Exception $e)
+  {
+     var_dump( $e );
+     exit;
+  }
+}
+*/
+
+
+/**
+ * Connect to the database specified in MsSetup.inc; Column Encryption keywords automatically added when $keystore is not none
+ * @param  string  $keywords : string to append to the dsn string in PDO::_construct
+ * @param  array  $options : attributes to pass to PDO::_construct
+ * @param  bool  $disableCE : flag for disabling column encryption even when keystore is NOT none
+ *                          for testing fetching encrypted data when connection column encryption is off
+ * @return PDO connection object
+ */
+function connect($keywords='', $options=array(), $disableCE = false)
+{
+    try {
+        // simply use $databaseName from MsSetup.inc to facilitate testing in Azure,
+        // which does not support switching databases
+        require  'MsSetup.inc';
+        $dsn = "sqlsrv:Server=$server;database=$databaseName;ConnectionPooling=false;";
+        if ($keystore != "none" && !$disableCE) {
+            $dsn .= "ColumnEncryption=Enabled;";
+        }
+        if ($keystore == "ksp" && !$disableCE) {
+            require('AE_Ksp.inc');
+            $ksp_path = getKSPPath();
+            $dsn .= "CEKeystoreProvider=$ksp_path;CEKeystoreName=$ksp_name;CEKeystoreEncryptKey=$encrypt_key;";
+        }
+        if ($keywords) {
+            $dsn .= $keywords;
+        }
+        $conn = new PDO($dsn, $uid, $pwd, $options);
+        $conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+        return $conn;
```

Comment:
I'm adding a parameter that allows the user to set the errmode. default is exception.

Reply:
The more I think about this the more I'm hesitant of having error mode set in connect(). First and foremost, connect is connect and error mode is error mode. They are two separate calls which shouldn't be combined. One function should only have one major purpose, right?

Besides, you use string comparisons to ensure error checking of error modes, which slows down unnecessarily (albeit only a little bit) because the user can easily set error mode after getting the connection resource object. 

If you insist, then please use constants like PDO::ERRMODE_WARNING, PDO::ERRMODE_EXCEPTION and PDO::ERRMODE_SILENT instead of string constants.